app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt

```diff
@@ -74,8 +81,27 @@ class DRecordDetailsFragment : Fragment() {
             putBoolean("isSubFragment", true)
         }
 
+        var currentContact: Contact? = null
+        try {
+            currentContact = contactsViewModel.contacts.value?.first { c -> c.username == rec.clientID }
+        } catch (e: NoSuchElementException) {
+            // Handle the exception here
+            Log.e("Error", "Could not find contact with ID ${rec.clientID}")
+        }
+
+        if (currentContact != null) {
+            mapsViewModel.addRoute(
+                Route(
+                    0.0, //Should be the current location
+                    0.0,
+                    currentContact.latitude ?: 0.0,
+                    currentContact.longitude ?: 0.0
```

Comment:
Here it could be better to display no point on the map when the `currentContact` hasn't some Coordinates set